### PR TITLE
Improve parse_output_projection

### PR DESCRIPTION
**Pull request by jimboH**
#88 Deal with the first improvement.
It seems like the second one has already been implemented.

### CODE REVIEW

File: reproject/utils.py [hunk 1]
```diff
@@ -44,7 +44,15 @@ def parse_output_projection(output_projection, shape_out=None):
         wcs_out = output_projection
         if shape_out is None:
             raise ValueError("Need to specify shape when specifying output_projection as WCS object")
+    elif isinstance(output_projection, string_type):
```
Comment (astrofrog) [inline reproject/utils.py]: you can replace ``string_type`` by ``six.string_types``

File: reproject/utils.py [hunk 2]
```diff
@@ -44,7 +44,15 @@ def parse_output_projection(output_projection, shape_out=None):
...
+        shape_out = hdu_list[0].data.shape
+        header = hdu_list[0].header
+        wcs_out = WCS(header)
+        hdu_list.close()
     else:
-        raise TypeError('output_projection should either be a Header or a WCS object')
+        raise TypeError('output_projection should either be a Header, a WCS object, or a filename')
 
+    if len(shape_out) == 0:
+        raise ValueError("The shape of the output header should not be an empty tuple")
```
Comment (astrofrog) [inline reproject/utils.py]: output header -> output image

**Comment by jimboH** [comment]
@astrofrog yes I would like to add entry to CHANGES.md. Under what milestone should I add the entry? (0.5?)

File: reproject/utils.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import numpy as np
-import sys
+from six import string_types
```
Comment (astrofrog) [inline reproject/utils.py]: six is already imported below:

```
 from astropy.extern import six
```

so just use ``six.string_types`` and don't add an import

Comment (jimboH) [reply reproject/utils.py]: Thank you! I missed that line of code...

**Comment by astrofrog** [comment]
Yes 0.5 is the correct place to add it, thanks!

**Pull request merged**